### ISSUE TITLE
partition is slow

### ISSUE THREAD

**Issue by ldionne** [ericniebler/meta#52]
Metabench is timing out because of it: https://travis-ci.org/ldionne/metabench/jobs/294051656#L1991

Also, graphs agree with this: http://metaben.ch/type/clang++-4.0/partition/index.html

It's fine if there's no desire to make it faster, but I'll reduce its range from Metabench to avoid timing out. Just let me know.

**Comment by gnzlbg** [comment]
@ldionne PR welcome :D

**Comment by gnzlbg** [comment]
This doesn't make metabench time out any more thanks to @ldionne 's patch but partition in meta still seems to be very slow compared to the other libraries.

**Comment by ldionne** [comment]
I'm sorry, I don't have the bandwidth to provide a PR for fixing this at the root in meta. That's why I just put a band-aid by fixing metabench. Also, FWIW, this seems to be an interaction between meta and GCC 7, since I think partition is not as slow on other compilers. There may be something interesting to report to the GCC folks.

Anyway, if this is fixed properly in meta and you want me to re-increase the range in metabench, just LMK.

**Comment by CaseyCarter** [comment]
> I'm sorry, I don't have the bandwidth to provide a PR for fixing this at the root in meta.

Re-opening the issue is not a suggestion that *you* should fix it, but that *someone* probably should.

**Comment by ldionne** [comment]
I understand that; I was rationalizing my initial closing of the issue.

**Comment by gnzlbg** [comment]
Yes I did not meant to say that you should fix it. I just don't want to "forget" that this happens. Thanks for reporting this!
